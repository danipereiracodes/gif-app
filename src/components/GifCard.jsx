import PropTypes from 'prop-types'

const GifCard = ({ title, url, id }) => {
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
            
        </div>
    )
}

GifCard.propTypes = {

    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default GifCard