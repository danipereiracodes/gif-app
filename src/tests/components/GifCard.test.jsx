import { render, screen } from "@testing-library/react";
import GifCard from "../../components/GifCard";

describe('Testing <GifCard />', () => {

    const key = 1;
    const gifs = {
        title: 'Image title',
        url: 'http://imagetest.com/',
    }

    test('Must match snapshot', () => {

        const { container } = render( <GifCard key={key} {...gifs} />)
        expect( container ).toMatchSnapshot()

       
        
    });

    test('Must img with the url and alt text', () => {

        render( <GifCard key={key} {...gifs} />)
        
        const { src , alt } = screen.getByRole('img')
        
        expect( src ).toBe( gifs.url )
        expect( alt ).toBe( gifs.title )

       
        
    });

    test('Must show title in our component', () => {

        render( <GifCard key={key} {...gifs} />)
        
        expect( screen.getByText( gifs.title )).toBeTruthy()

       
        
    });
    

});
