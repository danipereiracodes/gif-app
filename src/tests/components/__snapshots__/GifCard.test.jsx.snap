// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifCard /> Must match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Image title"
      src="http://imagetest.com/"
    />
    <p>
      Image title
    </p>
  </div>
</div>
`;
